### DISCUSSION TITLE
Update Wezterm title when the active Powershell pane changes or working directory is changed within a pane

### DISCUSSION THREAD

**Discussion by fcbry** [wezterm/wezterm#4805] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20231203-124028-e3cd2e93

### Ask your question!

I'm using starship and I've got the following snippet in my powershell profile.
```
$prompt = ""
function Invoke-Starship-PreCommand {
    $current_location = $executionContext.SessionState.Path.CurrentLocation
    if ($current_location.Provider.Name -eq "FileSystem") {
        $ansi_escape = [char]27
        $provider_path = $current_location.ProviderPath -replace "\\", "/"
        $prompt = "$ansi_escape]7;file://${env:COMPUTERNAME}/${provider_path}$ansi_escape\"
    }
    $host.ui.Write($prompt)
}

Invoke-Expression (&starship init powershell)
```
Everything works great, Wezterm is aware of the current pane directory, evidenced by the fact that new panes are launched in the same directory as the current active pane. 

What I can't figure out though is how to change the Wezterm window title when the active Powershell pane changes or working directory is changed within a pane, currently the Wezterm window title just displays pwsh.exe. 

I'm using some time tracking software, and one of the ways it works out what I'm doing is by tracking the window title. 

I've read a lot of posts, docs, and code but I'm relatively new to Wezterm/Lua so a bit out of my depth. 

Any help appreciated.

**Comment by wez** [comment]
You're using the OSC 7 escape sequence to indicate the working directory. 
You can use OSC 0, 1 or 2 to change the title information; see https://wezfurlong.org/wezterm/escape-sequences.html#operating-system-command-sequences

**Reply by wez** [reply, marked as answer]
alternatively, look at https://wezfurlong.org/wezterm/config/lua/window-events/format-tab-title.html or https://wezfurlong.org/wezterm/config/lua/window/set_right_status.html

**Reply by fcbry** [reply]
Thanks Wez, also, "Love your work!".
